src/utils/configUtils.ts

```diff
@@ -103,3 +135,149 @@ async function askPreferenceForConfig(configs: IExecutionConfig[], selectedConfi
 export function randomSequence(): string {
     return crypto.randomBytes(3).toString('hex');
 }
+
+type ApplyOperator = (value1: unknown, value2?: unknown) => boolean
+
+interface Token {
+    stringValue: string
+    getValue: () => unknown
+}
+
+interface ResultToken extends Token {
+    getValue: () => boolean
+}
+
+export class WhenClauseEvaluationContext {
+
+    private static readonly OPERATORS: Record<string, ApplyOperator> = {
+        // logical
+        '!': (value: unknown) => !value,
+        '&&': (value1: unknown, value2: unknown) => !!(value1 && value2),
+        '||': (value1: unknown, value2: unknown) => !!(value1 || value2),
+
+        // equality
+        '==': (value1: unknown, value2: unknown) => value1 === value2,
+        '===': (value1: unknown, value2: unknown) => value1 === value2,
+        '!=': (value1: unknown, value2: unknown) => value1 !== value2,
+        '!==': (value1: unknown, value2: unknown) => value1 !== value2,
+
+        // comparison
+        '>': (value1: number, value2: number) => value1 > value2,
+        '>=': (value1: number, value2: number) => value1 >= value2,
+        '<': (value1: number, value2: number) => value1 < value2,
+        '<=': (value1: number, value2: number) => value1 <= value2,
+
+        // match
+        '=~': (value: string, pattern: RegExp) => pattern.test(value),
+    }
+
+    private readonly context: Record<string, unknown> = {};
+
+    public constructor(readonly clause: string) {}
+
+    private tokenize(): Token[] {
+        const tokens: string[] = this.clause.split(/\s+/)
+            .flatMap((token: string) => token.split(/([\(\)]|!(?!=))/))
```

Comment:
I guess `[()]` can be used instead of `[\(\)]`?

Reply:
https://github.com/microsoft/vscode-java-test/pull/1638/commits/d73612c601c048d3398cb98f0b4127473dd14c76